File: frontend/src/interfaces/dto/adminrequests/AdminRequestDto.ts
import { DateTime } from 'luxon';

import User from '../../models/User';
import { Status } from '../../models/Status';

export interface AdminRequestDto {
    id: number;
    user: User;
    status: Status;
    approvedBy?: User;
    approvalDate?: Date;
    createdAt: DateTime;
}

File: src/main/java/org/itmo/isLab1/adminrequests/dto/AdminRequestDto.java
package org.itmo.isLab1.adminrequests.dto;

import lombok.Data;
import org.itmo.isLab1.adminrequests.Status;
import org.itmo.isLab1.common.framework.dto.ClientDto;
import org.itmo.isLab1.users.User;

import java.time.ZonedDateTime;

@Data
public class AdminRequestDto implements ClientDto{
    private int id;
    private User user;
    private Status status;
    private User approvedBy;
    private ZonedDateTime approvalDate;
    private ZonedDateTime createdAt;
}
